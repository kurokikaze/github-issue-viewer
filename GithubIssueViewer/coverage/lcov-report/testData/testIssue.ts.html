
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for testData/testIssue.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> / <a href="index.html">testData</a> testIssue.ts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {GithubIssueResponse} from '../types';
&nbsp;
export const testIssue: GithubIssueResponse = {
  url: 'https://api.github.com/repos/github/backup-utils/issues/770',
  repository_url: 'https://api.github.com/repos/github/backup-utils',
  labels_url:
    'https://api.github.com/repos/github/backup-utils/issues/770/labels{/name}',
  comments_url:
    'https://api.github.com/repos/github/backup-utils/issues/770/comments',
  events_url:
    'https://api.github.com/repos/github/backup-utils/issues/770/events',
  html_url: 'https://github.com/github/backup-utils/pull/770',
  id: 987963032,
  node_id: 'MDExOlB1bGxSZXF1ZXN0NzI2OTcxMTc5',
  number: 770,
  title: 'Clear GitHub Connect settings when not restoring settings',
  user: {
    login: 'lildude',
    id: 627280,
    node_id: 'MDQ6VXNlcjYyNzI4MA==',
    avatar_url: 'https://avatars.githubusercontent.com/u/627280?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/lildude',
    html_url: 'https://github.com/lildude',
    followers_url: 'https://api.github.com/users/lildude/followers',
    following_url:
      'https://api.github.com/users/lildude/following{/other_user}',
    gists_url: 'https://api.github.com/users/lildude/gists{/gist_id}',
    starred_url: 'https://api.github.com/users/lildude/starred{/owner}{/repo}',
    subscriptions_url: 'https://api.github.com/users/lildude/subscriptions',
    organizations_url: 'https://api.github.com/users/lildude/orgs',
    repos_url: 'https://api.github.com/users/lildude/repos',
    events_url: 'https://api.github.com/users/lildude/events{/privacy}',
    received_events_url: 'https://api.github.com/users/lildude/received_events',
    type: 'User',
    site_admin: true,
  },
  labels: [],
  state: 'open',
  locked: false,
  assignee: null,
  assignees: [],
  milestone: null,
  comments: 0,
  created_at: '2021-09-03T17:12:00Z',
  updated_at: '2021-09-08T16:27:33Z',
  closed_at: null,
  author_association: 'MEMBER',
  active_lock_reason: null,
  pull_request: {
    url: 'https://api.github.com/repos/github/backup-utils/pulls/770',
    html_url: 'https://github.com/github/backup-utils/pull/770',
    diff_url: 'https://github.com/github/backup-utils/pull/770.diff',
    patch_url: 'https://github.com/github/backup-utils/pull/770.patch',
  },
  body: "At the moment, when a backup it restored _without_ any settings, ie without using the `--config` flag, the GitHub Connect settings are still restored. This is because they're stored in the MySQL database and there's nothing in place to clear them or prevent them from being restored.\r\n\r\nThis leads to a problem where the restored instance will have all the same settings and GitHub Connect-specific unique ID as the original host and will start competing with the original instance if both are active. This can result in one instance being kicked off GitHub Connect, or confusing license sync information, if both remain connected.\r\n\r\nThis PR resolves this problem by calling a new script immediately after the MySQL database has been restored. This script will be added to all currently supported versions of GHES.",
  performed_via_github_app: null,
};
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Sep 09 2021 02:06:51 GMT+0300 (Москва, стандартное время)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    