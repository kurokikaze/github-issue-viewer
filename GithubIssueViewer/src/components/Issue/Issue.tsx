import React, {useContext} from 'react';
import {formatDistanceToNow} from 'date-fns';
import {
  View,
  Text,
  Button,
  StyleProp,
  ViewStyle,
  TouchableOpacity,
} from 'react-native';
import {GithubIssueResponse} from '../../types';

import styles from './styles';
import {ThemeContext} from '../ThemeContext/ThemeContext';

type IssueProps = {
  issue: GithubIssueResponse;
  onSelect: () => void;
  onBookmark?: () => void;
  onRemoveBookmark?: () => void;
  canBookmark?: boolean;
  style?: StyleProp<ViewStyle>;
};

export const Issue = ({
  issue,
  onSelect,
  onBookmark,
  canBookmark = false,
  onRemoveBookmark,
  style,
}: IssueProps) => {
  const theme = useContext(ThemeContext);

  return (
    <View style={[styles.issueContainer, style]}>
      <TouchableOpacity onPress={() => onSelect()}>
        <Text style={styles.issueTitle}>{issue.title}</Text>
        <Text style={[theme.textStyle, styles.issueDate]}>
          Created {formatDistanceToNow(new Date(issue.created_at))} ago
        </Text>
        {issue.updated_at && (
          <Text style={[theme.textStyle, styles.issueDate]}>
            Updated {formatDistanceToNow(new Date(issue.updated_at))} ago
          </Text>
        )}
      </TouchableOpacity>
      {canBookmark && onBookmark ? (
        <View style={styles.bookmarkButtonContainer}>
          <Button title="Add to bookmarks" onPress={() => onBookmark()} />
        </View>
      ) : null}
      {!canBookmark && onRemoveBookmark ? (
        <View style={styles.bookmarkButtonContainer}>
          <Button title="Remove bookmark" onPress={() => onRemoveBookmark()} />
        </View>
      ) : null}
    </View>
  );
};
