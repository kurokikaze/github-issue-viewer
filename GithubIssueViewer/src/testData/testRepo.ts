import {GithubRepositoryResponse, GithubUserResponse} from '../types';

const testOwner: GithubUserResponse = {
  login: 'kurokikaze',
  id: 92214,
  node_id: 'MDQ6VXNlcjkyMjE0',
  avatar_url: 'https://avatars.githubusercontent.com/u/92214?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/kurokikaze',
  html_url: 'https://github.com/kurokikaze',
  followers_url: 'https://api.github.com/users/kurokikaze/followers',
  following_url:
    'https://api.github.com/users/kurokikaze/following{/other_user}',
  gists_url: 'https://api.github.com/users/kurokikaze/gists{/gist_id}',
  starred_url: 'https://api.github.com/users/kurokikaze/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/kurokikaze/subscriptions',
  organizations_url: 'https://api.github.com/users/kurokikaze/orgs',
  repos_url: 'https://api.github.com/users/kurokikaze/repos',
  events_url: 'https://api.github.com/users/kurokikaze/events{/privacy}',
  received_events_url:
    'https://api.github.com/users/kurokikaze/received_events',
  type: 'User',
  site_admin: false,
};

export const testRepo: GithubRepositoryResponse = {
  id: 215071064,
  node_id: 'MDEwOlJlcG9zaXRvcnkyMTUwNzEwNjQ=',
  name: 'amadeus-bot',
  full_name: 'kurokikaze/amadeus-bot',
  private: false,
  owner: testOwner,
  html_url: 'https://github.com/kurokikaze/amadeus-bot',
  description: 'Bot for "Unleash the Geek" challenge',
  fork: false,
  url: 'https://api.github.com/repos/kurokikaze/amadeus-bot',
  forks_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/forks',
  keys_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/keys{/key_id}',
  collaborators_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/collaborators{/collaborator}',
  teams_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/teams',
  hooks_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/hooks',
  issue_events_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/issues/events{/number}',
  events_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/events',
  assignees_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/assignees{/user}',
  branches_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/branches{/branch}',
  tags_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/tags',
  blobs_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/blobs{/sha}',
  git_tags_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/tags{/sha}',
  git_refs_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/refs{/sha}',
  trees_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/trees{/sha}',
  statuses_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/statuses/{sha}',
  languages_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/languages',
  stargazers_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/stargazers',
  contributors_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/contributors',
  subscribers_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/subscribers',
  subscription_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/subscription',
  commits_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/commits{/sha}',
  git_commits_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/git/commits{/sha}',
  comments_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/comments{/number}',
  issue_comment_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/issues/comments{/number}',
  contents_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/contents/{+path}',
  compare_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/compare/{base}...{head}',
  merges_url: 'https://api.github.com/repos/kurokikaze/amadeus-bot/merges',
  archive_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/{archive_format}{/ref}',
  downloads_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/downloads',
  issues_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/issues{/number}',
  pulls_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/pulls{/number}',
  milestones_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/milestones{/number}',
  notifications_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/notifications{?since,all,participating}',
  labels_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/labels{/name}',
  releases_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/releases{/id}',
  deployments_url:
    'https://api.github.com/repos/kurokikaze/amadeus-bot/deployments',
  created_at: '2019-10-14T14:52:50Z',
  updated_at: '2019-10-16T08:45:32Z',
  pushed_at: '2019-10-16T08:45:30Z',
  git_url: 'git://github.com/kurokikaze/amadeus-bot.git',
  ssh_url: 'git@github.com:kurokikaze/amadeus-bot.git',
  clone_url: 'https://github.com/kurokikaze/amadeus-bot.git',
  svn_url: 'https://github.com/kurokikaze/amadeus-bot',
  homepage: null,
  size: 14,
  stargazers_count: 0,
  watchers_count: 0,
  language: 'JavaScript',
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: false,
  forks_count: 0,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 0,
  license: null,
  forks: 0,
  open_issues: 0,
  watchers: 0,
  default_branch: 'master',
};

export const anotherTestRepo: GithubRepositoryResponse = {
  id: 155540694,
  node_id: 'MDEwOlJlcG9zaXRvcnkxNTU1NDA2OTQ=',
  name: 'andor-vis',
  full_name: 'kurokikaze/andor-vis',
  private: false,
  owner: testOwner,
  html_url: 'https://github.com/kurokikaze/andor-vis',
  description: 'Andor map graph and visualisation',
  fork: false,
  url: 'https://api.github.com/repos/kurokikaze/andor-vis',
  forks_url: 'https://api.github.com/repos/kurokikaze/andor-vis/forks',
  keys_url: 'https://api.github.com/repos/kurokikaze/andor-vis/keys{/key_id}',
  collaborators_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/collaborators{/collaborator}',
  teams_url: 'https://api.github.com/repos/kurokikaze/andor-vis/teams',
  hooks_url: 'https://api.github.com/repos/kurokikaze/andor-vis/hooks',
  issue_events_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/issues/events{/number}',
  events_url: 'https://api.github.com/repos/kurokikaze/andor-vis/events',
  assignees_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/assignees{/user}',
  branches_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/branches{/branch}',
  tags_url: 'https://api.github.com/repos/kurokikaze/andor-vis/tags',
  blobs_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/blobs{/sha}',
  git_tags_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/tags{/sha}',
  git_refs_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/refs{/sha}',
  trees_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/trees{/sha}',
  statuses_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/statuses/{sha}',
  languages_url: 'https://api.github.com/repos/kurokikaze/andor-vis/languages',
  stargazers_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/stargazers',
  contributors_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/contributors',
  subscribers_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/subscribers',
  subscription_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/subscription',
  commits_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/commits{/sha}',
  git_commits_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/git/commits{/sha}',
  comments_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/comments{/number}',
  issue_comment_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/issues/comments{/number}',
  contents_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/contents/{+path}',
  compare_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/compare/{base}...{head}',
  merges_url: 'https://api.github.com/repos/kurokikaze/andor-vis/merges',
  archive_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/{archive_format}{/ref}',
  downloads_url: 'https://api.github.com/repos/kurokikaze/andor-vis/downloads',
  issues_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/issues{/number}',
  pulls_url: 'https://api.github.com/repos/kurokikaze/andor-vis/pulls{/number}',
  milestones_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/milestones{/number}',
  notifications_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/notifications{?since,all,participating}',
  labels_url: 'https://api.github.com/repos/kurokikaze/andor-vis/labels{/name}',
  releases_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/releases{/id}',
  deployments_url:
    'https://api.github.com/repos/kurokikaze/andor-vis/deployments',
  created_at: '2018-10-31T10:42:36Z',
  updated_at: '2018-10-31T12:18:29Z',
  pushed_at: '2018-10-31T12:18:28Z',
  git_url: 'git://github.com/kurokikaze/andor-vis.git',
  ssh_url: 'git@github.com:kurokikaze/andor-vis.git',
  clone_url: 'https://github.com/kurokikaze/andor-vis.git',
  svn_url: 'https://github.com/kurokikaze/andor-vis',
  homepage: null,
  size: 2,
  stargazers_count: 0,
  watchers_count: 0,
  language: 'JavaScript',
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: false,
  forks_count: 0,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 5,
  license: null,
  forks: 0,
  open_issues: 0,
  watchers: 0,
  default_branch: 'master',
};
