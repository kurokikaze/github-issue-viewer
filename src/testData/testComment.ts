import {testOwner} from './testRepo';

export const testComment = {
  url: 'https://api.github.com/repos/github/backup-utils/issues/comments/899664665',
  html_url:
    'https://github.com/github/backup-utils/issues/761#issuecomment-899664665',
  issue_url: 'https://api.github.com/repos/github/backup-utils/issues/761',
  id: 899664665,
  node_id: 'IC_kwDOAUntRM41n8sZ',
  user: testOwner,
  created_at: '2021-08-16T16:56:14Z',
  updated_at: '2021-08-16T16:56:14Z',
  author_association: 'CONTRIBUTOR',
  body: 'Comment body',
  performed_via_github_app: null,
};
