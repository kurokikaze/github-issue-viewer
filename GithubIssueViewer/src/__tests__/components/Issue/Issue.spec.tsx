import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import Issue from '../../../components/Issue/Issue';
import {testIssue} from '../../../testData/testIssue';
import {formatDistanceToNow} from 'date-fns';

jest.mock('date-fns');

const mockDistanceFormatter = formatDistanceToNow as jest.MockedFunction<
  typeof formatDistanceToNow
>;

mockDistanceFormatter.mockReturnValue('some time ago');

describe('Issue', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(
      <Issue
        issue={testIssue}
        onSelect={() => null}
        onBookmark={() => null}
        onRemoveBookmark={() => null}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly (can bookmark)', () => {
    const wrapper = renderer.create(
      <Issue
        issue={testIssue}
        onSelect={() => null}
        onBookmark={() => null}
        onRemoveBookmark={() => null}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('onSelect', () => {
    const issueSelectHandler = jest.fn();

    const wrapper = renderer.create(
      <Issue
        issue={testIssue}
        onSelect={issueSelectHandler}
        onBookmark={() => null}
        onRemoveBookmark={() => null}
      />,
    );

    wrapper.root.findByProps({'data-testID': 'select-issue'}).props.onPress();

    expect(issueSelectHandler).toHaveBeenCalled();
  });

  it('onBookmark', () => {
    const issueBookmarkHandler = jest.fn();

    const wrapper = renderer.create(
      <Issue
        issue={testIssue}
        onSelect={() => null}
        canBookmark
        onBookmark={issueBookmarkHandler}
        onRemoveBookmark={() => null}
      />,
    );

    wrapper.root.findByProps({'data-testID': 'bookmark-issue'}).props.onPress();

    expect(issueBookmarkHandler).toHaveBeenCalled();
  });

  it('onRemoveBookmark', () => {
    const issueRemoveBookmarkHandler = jest.fn();

    const wrapper = renderer.create(
      <Issue
        issue={testIssue}
        onSelect={() => null}
        onBookmark={() => null}
        onRemoveBookmark={issueRemoveBookmarkHandler}
      />,
    );

    wrapper.root
      .findByProps({'data-testID': 'remove-issue-from-bookmarks'})
      .props.onPress();

    expect(issueRemoveBookmarkHandler).toHaveBeenCalled();
  });
});
