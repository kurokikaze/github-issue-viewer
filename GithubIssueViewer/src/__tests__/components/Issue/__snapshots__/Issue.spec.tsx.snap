// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue renders correctly (can bookmark) 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#fff",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "paddingBottom": 5,
        "paddingTop": 18,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#056aff",
          "display": "flex",
          "flex": 1,
          "flexWrap": "wrap",
          "fontSize": 20,
          "fontWeight": "normal",
          "lineHeight": 24,
          "marginBottom": 20,
          "paddingLeft": 20,
        }
      }
    >
      Clear GitHub Connect settings when not restoring settings
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {
            "fontSize": 18,
            "lineHeight": 20,
            "paddingLeft": 20,
          },
        ]
      }
    >
      Created 
      some time ago
       ago
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {
            "fontSize": 18,
            "lineHeight": 20,
            "paddingLeft": 20,
          },
        ]
      }
    >
      Updated 
      some time ago
       ago
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Remove bookmark
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Issue renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#fff",
        "borderBottomWidth": 1,
        "display": "flex",
        "flex": 1,
        "paddingBottom": 5,
        "paddingTop": 18,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#056aff",
          "display": "flex",
          "flex": 1,
          "flexWrap": "wrap",
          "fontSize": 20,
          "fontWeight": "normal",
          "lineHeight": 24,
          "marginBottom": 20,
          "paddingLeft": 20,
        }
      }
    >
      Clear GitHub Connect settings when not restoring settings
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {
            "fontSize": 18,
            "lineHeight": 20,
            "paddingLeft": 20,
          },
        ]
      }
    >
      Created 
      some time ago
       ago
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#333",
          },
          Object {
            "fontSize": 18,
            "lineHeight": 20,
            "paddingLeft": 20,
          },
        ]
      }
    >
      Updated 
      some time ago
       ago
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Remove bookmark
        </Text>
      </View>
    </View>
  </View>
</View>
`;
